### PR TITLE
[Inductor] Fix Conv Binary Inplace Fusion issue

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115809
* __->__ #115153

**Summary**
Take this Pattern as example
```
  #      ReLU
  #     /   \
  #  Conv1
  #   /      \
  # Conv2
  #   \      /
  #      Add
```
The current `ConvBinaryInplace` check will fail to perform Inplace fusion (using outplace fusion instead) due to `ReLU` having 2 users. However, if all users of `ReLU` are ancestor nodes of `Conv2`, we should be able to proceed with the `ConvBinaryInplace` fusion. This diff relaxes the `ConvBinaryInplace` check accordingly.

**TestPlan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_conv2d_binary_inplace_fusion_pass_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_conv2d_binary_inplace_fusion_failed_cpu
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by CaoE — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
@jgong5 Could you help to take a look of this PR?

File: torch/_inductor/fx_passes/mkldnn_fusion.py [hunk 1]
```diff
@@ -366,12 +366,67 @@ def fn(match):
...
+                #     /   \
+                #  Conv1
+                #   /      \
+                # Conv2
+                #   \      /
+                #      Add
+                # Although, the extra input node (ReLU) has more than 1 users: Conv1 and Add.
+                # The Conv1 is the ancestor node of the current compute node (Conv2).
+                # This indicates that the buffer of ReLU has completed all its usage,
+                # So we can safely make changes to it now by doing Conv2->Add inplace fusion.
```
[the remark this answers is not in the document]
Comment (leslie-fang-intel) [reply torch/_inductor/fx_passes/mkldnn_fusion.py]: Thanks for the comment and changed accordingly. Please help to take a look again.

**Review by jgong5 — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
@eellison Could you help to take a look of this PR?

**Comment by leslie-fang-intel** [comment]
Hi @eellison, could you kindly help to take a look of this PR? it may need your approve to be landed.

**Pull request closed without merging**